test/e2e/framework/pod/delete.go

```diff
@@ -39,6 +39,12 @@ const (
 func DeletePodOrFail(c clientset.Interface, ns, name string) {
 	ginkgo.By(fmt.Sprintf("Deleting pod %s in namespace %s", name, ns))
 	err := c.CoreV1().Pods(ns).Delete(context.TODO(), name, metav1.DeleteOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return
+		}
+	}
```

Comment:
```suggestion
	if apierrors.IsNotFound(err) {
 		return
	}
```

I believe the out `err != nil` check can be omitted :+1:

Reply:
That isn't documented, is it? It just follows from how https://github.com/kubernetes/apimachinery/blob/ab1231685bfe66237a116092641da00923cc00ca/pkg/api/errors/errors.go#L641-L648 is implemented.

I prefer defensive programming and thus keeping the nil check. The two if statements can be combined, though.

Reply:
The other functions below could also be simplified; a clean approach would be to first fix that in one commit, then add the same code here in a second one.

Reply:
the second function below already tolerates `pod not found` errors by the looks of it.